test/functional/sessions_controller_test.rb

```diff
@@ -111,7 +111,25 @@ class SessionsControllerTest < ActionController::TestCase
         assert_nil @controller.session[:mfa_user]
       end
     end
-  end
+
+    context "when no mfa_expires_at session is present" do
+      setup do
+        @controller.session[:mfa_user] = @user.id
+        travel 30.minutes
```

Comment:
I think we should consider either adding a `teardown` to remove the time travel, or wrap code in a `travel_to` block.